src/com/company/Shop1.java

```diff
@@ -1,88 +1,94 @@
 package com.company;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 
-public class Shop1 implements Shop{
+public class Shop1 implements Shop {
     private List<Product> productList = new ArrayList<>();
-    private Map<String, Integer> check = new HashMap();
+    private Map<Product, Integer> storageMap = new HashMap();
+    private Map<Long, Product> mapOfId = new HashMap<>();
+    private Map<Product, Integer> mapOfShoppingCart = new HashMap<>();
```

Comment:
why your shop stores shopping cart?
what if you have multiple buyers?
is shopping cart really a part of the shop?